Documentation/contributing/development/contributing_guide.rst

```diff
@@ -14,29 +14,31 @@ Clone and Provision Environment
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 #. Make sure you have a `GitHub account <https://github.com/join>`_
-#. Clone the cilium repository into your ``GOPATH``.
+#. Fork the Cilium repository to your GitHub user or organization.
+#. Clone your ``${YOUR_GITHUB_USERNAME_OR_ORG}/cilium`` fork into your ``GOPATH``, and setup the base repository as ``upstream`` remote:
 
    ::
 
-      mkdir -p $GOPATH/src/github.com/cilium
-      cd $GOPATH/src/github.com/cilium
-      git clone https://github.com/cilium/cilium.git
+      mkdir -p ${GOPATH}/src/github.com/cilium
+      cd ${GOPATH}/src/github.com/cilium
+      git clone https://github.com/${YOUR_GITHUB_USERNAME_OR_ORG}/cilium.git
       cd cilium
+      git remote add upstream https://github.com/cilium/cilium.git
```

Comment:
```suggestion
      git clone git@github.com:${YOUR_GITHUB_USERNAME_OR_ORG}/cilium.git
      cd cilium
      git remote add upstream git@github.com:cilium/cilium.git
```

Reply:
Why would you use SSH for the `upstream` repo? Makes you type your password every time, when you just need to pull from a public repo, no?

Reply:
No, it doesn't. I just ran the following and didn't need to type my password once:
```
$ cd /tmp/
$ git clone git@github.com:pchaigno/strace
$ cd strace/
$ git remote add upstream git@github.com:strace/strace
$ git fetch upstream 
$ git pull upstream master
```

How do you push to repositories over HTTPS if you have double factor authentication enabled?

Reply:
For me it does :/ On both `fetch` and `pull`.

Why would you ever push to the upstream repo if you create the PR from your fork?

Reply:
> For me it does :/ On both `fetch` and `pull`.

Don't you have SSH keys configured? What about double factor auth.? I thought that was required for all Cilium members.

> Why would you ever push to the upstream repo if you create the PR from your fork?

Because we still have workflows that assume your PR branch is not in a fork. Probably fine for new contributors though. But if it works for `origin`, I don't see why it wouldn't for `upstream`.

Reply:
> Don't you have SSH keys configured?

Yes, and git asks me for the password to decrypt my key. You have no password on your SSH key? In that case I'd understand it needs none for you.

> Because we still have workflows that assume your PR branch is not in a fork. Probably fine for new contributors though. But if it works for `origin`, I don't see why it wouldn't for `upstream`.

My impression was indeed that this section was addressed at people starting to contribute, not to someone that would be involved in such workflow. Using SSH _works_ for `upstream`, it's just more painful when you have passwords on your SSH keys because you shouldn't need to type them for pulling from a public repo.

Reply:
> You have no password on your SSH key? In that case I'd understand it needs none for you.

I don't usually need to enter every single time I use my key ;-)

> My impression was indeed that this section was addressed at people starting to contribute, not to someone that would be involved in such workflow. Using SSH _works_ for `upstream`, it's just more painful when you have passwords on your SSH keys because you shouldn't need to type them for pulling from a public repo.

Makes sense, let's just change for `origin` then.